pkg/sql/crdb_internal.go

```diff
@@ -5087,6 +5086,70 @@ CREATE TABLE crdb_internal.statement_statistics (
 	},
 }
 
+// crdb_internal.statement_statistics view merges in-memory cluster statement statistics from
+// the crdb_internal.cluster_statement_statistics virtual table with persisted statement
+// statistics from the system table. Equivalent stats between tables are aggregated. This
+// view is primarily used to query statement stats info by date range.
+var crdbInternalStmtStatsView = virtualSchemaView{
+	schema: `
+CREATE VIEW crdb_internal.statement_statistics AS
+SELECT
+	aggregated_ts,
+	fingerprint_id,
+	transaction_fingerprint_id,
+	plan_hash,
+	app_name,
+	max(metadata) as metadata,
+	crdb_internal.merge_statement_stats(array_agg(statistics)),
+  max(sampled_plan),
+  aggregation_interval
+FROM (
+	SELECT
+			aggregated_ts,
+			fingerprint_id,
+			transaction_fingerprint_id,
+			plan_hash,
+			app_name,
+			metadata,
+			statistics,
+			sampled_plan,
+			aggregation_interval
+	FROM
+			crdb_internal.cluster_statement_statistics
+	UNION ALL
+			SELECT
+					aggregated_ts,
+					fingerprint_id,
+					transaction_fingerprint_id,
+					plan_hash,
+					app_name,
+					metadata,
+					statistics,
+					plan,
+					agg_interval
+			FROM
+					system.statement_statistics
```

Comment:
the suggestion is that we do a group by and aggregate here as well as above to avoid needing to read everything, I'm not sure if the limit pushdown will work perfectly, but hopefully.